### DISCUSSION TITLE
Spark DayTimeIntervalType support

### DISCUSSION THREAD

**Discussion by zhli1142015** [facebookincubator/velox#9888] [Q&A] [answered]
Hello,
I'm adding Spark function `SubtractDates` which result type is `DayTimeIntervalType`. Velox defines silimar internal type： `IntervalDayTimeType`.First I want to use it but I found although both types have a physical length of 8 bytes, but their semantics are different. Velox `IntervalDayTimeType` represents time interval in milliseconds. Spark `DayTimeIntervalType` represents time interval in microseconds.
Should I add a new custom type for Spark `DayTimeIntervalType` or map it to `IntervalDayTimeType`? 
CC @mbasmanova @rui-mo  @FelixYBW 
Thanks.
https://spark.apache.org/docs/latest/sql-ref-datatypes.html
![image](https://github.com/facebookincubator/velox/assets/10524738/ea587540-54e0-4bc5-81d6-8204a433f673)

**Comment by rui-mo** [comment]
I wonder if it is feasible to use micro unit in IntervalDayTimeType and provide toMillis method just like TimestampType does.

**Reply by zhli1142015** [reply]
In Velox, the Timestamp type represents a time instant with nanosecond precision. It should be acceptable to adjust the precision to microseconds or milliseconds. However, applying this approach to the IntervalDayTimeType might reduce the data representation range for Presto. I'm unsure if this is a problem.

**Comment by mbasmanova** [comment, marked as answer]
@zhli1142015 It looks like in Spark DayTimeIntervalType has parameters. It is not a singleton type. If so, I think you need to define a new type.

**Reply by zhli1142015** [reply]
Thanks.
